include/exadg/incompressible_navier_stokes/user_interface/parameters.cpp

```diff
@@ -433,6 +469,11 @@ Parameters::check(dealii::ConditionalOStream const & pcout) const
   // PRESSURE-CORRECTION SCHEME
   if(temporal_discretization == TemporalDiscretization::BDFPressureCorrection)
   {
+    AssertThrow(
+      spatial_discretization == SpatialDiscretization::L2,
+      dealii::ExcMessage(
+        "Pressure correction only works for L2. If you want to use Hdiv conforming elements you have to use the coupled solution"));
```

Comment:
I would do this differently. I would make sure by the Assert that the spatial discretization is not Hdiv. If it is Hdiv, the ExcMessage should be of type "not implemented", because this reflects better the current situation, right? Hdiv for pressure-correction has just not been implemented? Or do you think there are other special issues in case of Hdiv with pressure-correction?

Reply:
I just haven't tried it. It might have been the inverse mass that wasn't working, so it might work now. I'll still add the assert though.